iceoryx_hoofs/include/iceoryx_hoofs/design_pattern/creation.hpp

```diff
@@ -142,6 +142,13 @@ class Creation
     ErrorType m_errorValue;
 };
 
+template <typename Class, typename Error>
+void creationMove(Creation<Class, Error>& lhs, Creation<Class, Error>&& rhs) noexcept
+{
+    using CreationPattern_t = Creation<Class, Error>;
+    lhs.operator=(std::forward<CreationPattern_t>(rhs));
+}
```

Comment:
Please don't do it this way. With this you still have to do `creationMove(*this, std::move(rhs))` which caused the linter warnings or `creationMove(*this, std::move(*static_cast<Creation<Foo>*>(&rhs)))` which we want to prevent with this method.

Please rewrite this to
```cpp
void creationMove(Creation<Class, Error>& rhs)
{
    *this = std::move(rhs);
}
```

This might cause some warnings though. In this case, please use a macro.

Reply:
On a second thought, maybe we need to do this completely different https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-copy-virtual